### DISCUSSION TITLE
Create dbt exposures from Dashboards

### DISCUSSION THREAD

**Discussion by timhiebenthal** [lightdash/lightdash#3893] [Ideas]
Would be cool if the dbt integration would also work "backwards" [as it does Preset does](https://preset.io/blog/dbt-superset-integration-preset/#synchronizing-exposures-from-preset-back-to-dbt)

Alternatively adding the menu item `Export exposure.yml` in the Lightdash UI would be also a decent solution for non-technical analysts.

![image](https://user-images.githubusercontent.com/57394222/205124043-e14510e4-c39f-4bef-9d37-2a6caab003cf.png)

Expected output would be some pre-filled boilerplate.
name, owner, url, models etc. could be easily extracted I guess

What do you think?

**Comment by TuringLovesDeathMetal** [comment]
oooh neat idea! 

This is kind of similar to #369 (which is a big ole issue that probably has a lot of sub-issues). 

@timhiebenthal do you imagine that you'd be able to "sync" these two things? Or would it just be kind of a one-off export feature?

**Reply by timhiebenthal** [reply]
I guess it depends a bit on who the user is

**As a rather technical Analytics Engineer:**
I think adding it to the existing schema-generation flow might make sense from a workflow perspective
https://docs.lightdash.com/guides/cli/how-to-auto-generate-schema-files

Ideally I would create or adjust dashboards in the UI, hit a command (or having it part of the CI/CD) generating a new or modify the existing exposure.yml

**As the business analyst who builds dashboards but usually does do any coding** it might be rather part of my workflow to download this .yml-File from the UI after creating a new dashboard

But the goal is similar: I want up-to-date exposure.ymls in my dbt docs with the least effort possible; so I don't see this as a one-off export

**Reply by TuringLovesDeathMetal** [reply]
Cool, makes sense. So:
-  it should be synced. 
- And you should be able to sync this either from the UI or from the command line. 

Is that a fair summary?

**Reply by timhiebenthal** [reply]
Exactly, yeah :)

**Reply by ZeRego** [reply]
@TuringLovesDeathMetal fyi I opened an issue for this feature https://github.com/lightdash/lightdash/issues/4062. feel free to add more information/requirements there :)
